packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -231,6 +233,12 @@ export class Outbox {
 		}
 	}
 
+	public submitIdAllocation(message: BatchMessage) {
+		this.maybeFlushPartialBatch();
+
+		this.addMessageToBatchManager(this.idAllocationBatch, message);
```

Comment:
Yes, it should. 

But I think the other question is why are these special that they need their own separate batch manager? And is it intentional that they flush first? @justus-camp 

The main reason I'm asking is because we'd love to move away from this model. Initially, we wanted to alleviate the pain from large batches by separating regular ops vs attach ops (attach ops are huge) but with compression/chunking/grouping the point is moot right now. Except for blob attach ops which are special wrt to grouped batching, we'd like to NOT invest more in this area and actually merge these batch managers in the future in order to just simplify things.

Reply:
Unfortunately it's intentional that they flush first. In integrating the compressor into tree I discovered a bug where if a resubmitted op generates a new ID we can finalize ID ranges out of order (my memory is a little fuzzy on this, I opened this PR a while back). Moving the compressor ops into their own batch ensures that all ID ranges are finalized before any ops that are dependent on them are processed.